extensions/variants.go

```diff
@@ -140,24 +180,35 @@ type ScalarFunctionVariant struct {
 	impl        ScalarFunctionImpl
 }
 
-func (s *ScalarFunctionVariant) Name() string                     { return s.name }
-func (s *ScalarFunctionVariant) Description() string              { return s.description }
-func (s *ScalarFunctionVariant) Args() ArgumentList               { return s.impl.Args }
-func (s *ScalarFunctionVariant) Options() map[string]Option       { return s.impl.Options }
-func (s *ScalarFunctionVariant) Variadic() *VariadicBehavior      { return s.impl.Variadic }
-func (s *ScalarFunctionVariant) Deterministic() bool              { return s.impl.Deterministic }
-func (s *ScalarFunctionVariant) SessionDependent() bool           { return s.impl.SessionDependent }
-func (s *ScalarFunctionVariant) Nullability() NullabilityHandling { return s.impl.Nullability }
-func (s *ScalarFunctionVariant) URI() string                      { return s.uri }
+func (s *ScalarFunctionVariant) Name() string                { return s.name }
+func (s *ScalarFunctionVariant) Description() string         { return s.description }
+func (s *ScalarFunctionVariant) Args() ArgumentList          { return s.impl.Args }
+func (s *ScalarFunctionVariant) Options() map[string]Option  { return s.impl.Options }
+func (s *ScalarFunctionVariant) Variadic() *VariadicBehavior { return s.impl.Variadic }
+func (s *ScalarFunctionVariant) Deterministic() bool         { return s.impl.Deterministic }
+func (s *ScalarFunctionVariant) SessionDependent() bool      { return s.impl.SessionDependent }
+func (s *ScalarFunctionVariant) Nullability() NullabilityHandling {
+	if s.impl.Nullability == "" {
```

Comment:
we shouldn't need to handle this here. we should figure out a way to handle on unmarshal. I think something like github.com/creasty/defaults will allow us to use tags to handle defaults rather than having to hand code fixes. (Not sure if that is the best library, just an option).

Reply:
fixed it to use "github.com/creasty/defaults "